core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -82,6 +85,14 @@ public UnsafeSorterSpillReader(
       Closeables.close(bs, /* swallowIOException = */ true);
       throw e;
     }
+    taskContext.addTaskCompletionListener(context -> {
+      try {
+        close();
+      } catch (IOException e) {
+        logger.error("error while closing UnsafeSorterSpillReader", e);
+        throw new RuntimeException(e);
```

Comment:
We can log in `info` and drop the exception - for 'normal' flow, that is what ends up happening when close throws exception.